File: single-language-analysis/dataset-files/python/test_smell_103.py
# Original URL: https://github.com/wookayin/gpustat/blob/7a3b9644229681a752b560c6d4889929b9ea39d5/gpustat/test_gpustat.py#L707-L733

class TestGPUStat(object):

    def test_smell_103(self, scenario_basic):
        """Tests gpustat CLI with a variety of --show-xxx options. """

        capture_output = self.capture_output
        print('')

        TEST_OPTS = []
        TEST_OPTS += ['-a', '-c', '-u', '-p', '-e', '-P', '-f']
        TEST_OPTS += [('-e', ''), ('-P', '')]
        TEST_OPTS += [('-e', 'enc,dec'), '-Plimit,draw']
        TEST_OPTS += ['-cup', '-cpu', '-cufP']  # 'cpuePf'

        for opt in TEST_OPTS:
            if isinstance(opt, str):
                opt = [opt]

            print('\x1b[30m\x1b[43m',  # black_on_yellow
                  '$ gpustat ' + ' '.join(shlex.quote(o) for o in opt),
                  '\x1b(B\x1b[m', sep='')
            s = capture_output('gpustat', *opt)

            # TODO: Validate output without hardcoding expected outputs
            print(s)

        # Finally, unknown args
        with pytest.raises(AssertionError):
            capture_output('gpustat', '--unrecognized-args-in-test')

File: single-language-analysis/dataset-files/java/TestSmell103.java
import org.junit.Test;
import static org.junit.Assert.*;

public class TestSmell103 {

    @Test
    public void testArgsCommandlineShowOptions() {
        String captureOutput = this.captureOutput();
        System.out.println("");

        String[] testOpts = new String[]{
            "-a", "-c", "-u", "-p", "-e", "-P", "-f",
            "-e", "", "-P", "",
            "-e", "enc,dec", "-Plimit,draw",
            "-cup", "-cpu", "-cufP"
        };

        for (String opt : testOpts) {
            String[] options;
            if (opt instanceof String) {
                options = new String[]{opt};
            } else {
                options = (String[]) opt;
            }

            System.out.print("\u001B[30m\u001B[43m" + "$ gpustat " + String.join(" ", options) + "\u001B(B\u001B[m");
            String s = captureOutput("gpustat", options);

            // TODO: Validate output without hardcoding expected outputs
            System.out.println(s);
        }

        // Finally, unknown args
        try {
            captureOutput("gpustat", "--unrecognized-args-in-test");
            fail("Expected AssertionError");
        } catch (AssertionError e) {
            // Expected exception
        }
    }
}